src/abbfreeathome/devices/cover_actuator.py

```diff
@@ -0,0 +1,389 @@
+    """Free@Home BlindActuator Class."""
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        channel_id: str,
+        channel_name: str,
+        inputs: dict[str, dict[str, Any]],
+        outputs: dict[str, dict[str, Any]],
+        parameters: dict[str, dict[str, Any]],
+        api: FreeAtHomeApi,
+        floor_name: str | None = None,
+        room_name: str | None = None,
+    ) -> None:
+        """Initialize the Free@Home BindActuator class."""
+
+        super().__init__(
+            device_id,
+            device_name,
+            channel_id,
+            channel_name,
+            inputs,
+            outputs,
+            parameters,
+            api,
+            floor_name,
+            room_name,
+        )
+
+
+class ShutterActuator(CoverActuator):
+    """Free@Home ShutterActuator Class."""
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        channel_id: str,
+        channel_name: str,
+        inputs: dict[str, dict[str, Any]],
+        outputs: dict[str, dict[str, Any]],
+        parameters: dict[str, dict[str, Any]],
+        api: FreeAtHomeApi,
+        floor_name: str | None = None,
+        room_name: str | None = None,
+    ) -> None:
+        """Initialize the Free@Home ShutterActuator class."""
+        self._tilt_position: int | None = None
+
+        super().__init__(
+            device_id,
+            device_name,
+            channel_id,
+            channel_name,
+            inputs,
+            outputs,
+            parameters,
+            api,
+            floor_name,
+            room_name,
+        )
+
+    @property
+    def tilt_position(self) -> int | None:
+        """Get the tilt position of the cover."""
+        return self._tilt_position
+
+    async def set_tilt_position(self, value: int):
+        """
+        Set the tilt position of the cover.
+
+        The tilt position has to be between 0 and 100
+        Fully open = 0
+        Fully closed = 100
+        Just as an information: This is exaclty the other way round as done in HA,
+        so in HA we have to remember to convert the value with something like:
+        abs(value-100)
+        before sending it to this function
+        """
+        value = max(0, value)
+        value = min(value, 100)
+
+        await self._set_tilt_datapoint(str(value))
+        self._tilt_position = value
+
+    def _refresh_state_from_output(self, output: dict[str, Any]) -> bool:
+        """
+        Refresh the state of the device from a given output.
+
+        This will return whether the state was refreshed as a boolean value.
+        """
+        if super()._refresh_state_from_output(output) is True:
+            return True
+        if (
+            output.get("pairingID")
+            == Pairing.AL_CURRENT_ABSOLUTE_POSITION_SLATS_PERCENTAGE.value
+        ):
+            self._tilt_position = int(output.get("value"))
+            return True
```

Comment:
This can also be placed in the parent `CoverActuator` class.

Reply:
Because `AL_CURRENT_ABSOLUTE_POSITION_SLATS_PERCENTAGE` is defined in the class variable `_state_refresh_output_pairings` of the parent class  that means the parent class will attempt to refresh the state for this pairing id in the `CoverActuator` class. It would then make sense the parent class can also handle this logic. And if the parent class needs to handle this logic that means `self_.tilt_position` should also be in the parent class.

Reply:
I'll do it, but I'm not 100% happy with it. As we have the sub-class `ShutterActuator`, why not use it for the specific stuff and keep the `CoverActuator` as straight as possible?

Reply:
understand